libraries/AP_ExternalAHRS/AP_ExternalAHRS_VectorNav.cpp

```diff
@@ -351,6 +581,98 @@ void AP_ExternalAHRS_VectorNav::process_packet2(const uint8_t *b)
     AP::gps().handle_external(gps);
 }
 
+/*
+  process VN-100 packet type 1
+ */
+void AP_ExternalAHRS_VectorNav::process_packet_VN_100(const uint8_t *b)
+{
+    const struct VN_100_packet1 &pkt = *(struct VN_100_packet1 *)b;
+
+    last_pkt1_ms = AP_HAL::millis();
+
+    const bool use_uncomp = option_is_set(AP_ExternalAHRS::OPTIONS::VN_UNCOMP_IMU);
+
+    {
+        WITH_SEMAPHORE(state.sem);
+        if (use_uncomp) {
+            state.accel = Vector3f{pkt.uncompAccel[0], pkt.uncompAccel[1], pkt.uncompAccel[2]};
+            state.gyro = Vector3f{pkt.uncompAngRate[0], pkt.uncompAngRate[1], pkt.uncompAngRate[2]};
+        } else {
+            state.accel = Vector3f{pkt.accel[0], pkt.accel[1], pkt.accel[2]};
+            state.gyro = Vector3f{pkt.gyro[0], pkt.gyro[1], pkt.gyro[2]};
+        }
+
+        state.quat = Quaternion{pkt.quaternion[3], pkt.quaternion[0], pkt.quaternion[1], pkt.quaternion[2]};
+        state.have_quaternion = true;
+    }
+
+#if AP_BARO_EXTERNALAHRS_ENABLED
+    {
+        AP_ExternalAHRS::baro_data_message_t baro;
+        baro.instance = 0;
+        baro.pressure_pa = pkt.pressure*1e3;
+        baro.temperature = pkt.temp;
+
+        AP::baro().handle_external(baro);
+    }
+#endif
+
+    {
+        AP_ExternalAHRS::mag_data_message_t mag;
+        if (use_uncomp) {
+            mag.field = Vector3f{pkt.uncompMag[0], pkt.uncompMag[1], pkt.uncompMag[2]};
+        } else {
+            mag.field = Vector3f{pkt.mag[0], pkt.mag[1], pkt.mag[2]};
+        }
+        mag.field *= 1000; // to mGauss
+
+        AP::compass().handle_external(mag);
+    }
+
+    {
+        AP_ExternalAHRS::ins_data_message_t ins;
+
+        ins.accel = state.accel;
+        ins.gyro = state.gyro;
+        ins.temperature = pkt.temp;
+
+        AP::ins().handle_external(ins);
+    }
+
+    // @LoggerMessage: EAH3
+    // @Description: External AHRS data
+    // @Field: TimeUS: Time since system startup
+    // @Field: Temp: Temprature
+    // @Field: Pres: Pressure
+    // @Field: MX: Magnetic feild X-axis
+    // @Field: MY: Magnetic feild Y-axis
+    // @Field: MZ: Magnetic feild Z-axis
+    // @Field: AX: Acceleration X-axis
+    // @Field: AY: Acceleration Y-axis
+    // @Field: AZ: Acceleration Z-axis
+    // @Field: GX: Rotation rate X-axis
+    // @Field: GY: Rotation rate Y-axis
+    // @Field: GZ: Rotation rate Z-axis
+    // @Field: Q1: Attitude quaternion 1
+    // @Field: Q2: Attitude quaternion 2
+    // @Field: Q3: Attitude quaternion 3
+    // @Field: Q4: Attitude quaternion 4
+
+    AP::logger().WriteStreaming("EAH3", "TimeUS,Temp,Pres,MX,MY,MZ,AX,AY,AZ,GX,GY,GZ,Q1,Q2,Q3,Q4",
```

Comment:
All of this except the quaternion is logged in the IMU/MAG/BARO, but having a direct log is uesfull.